src/components/application_manager/src/policies/policy_handler.cc

```diff
@@ -1283,7 +1283,31 @@ void PolicyHandler::OnSnapshotCreated(const BinaryMessage& pt_string) {
     LOG4CXX_ERROR(logger_, "Service URLs are empty! NOT sending PT to mobile!");
     return;
   }
-  SendMessageToSDK(pt_string, urls.front().url.front());
+
+  static size_t current_app = 0;
```

Comment:
@anosach-luxoft 
On every OnSystemRequest retry SDL must change the URL it sends, walking sequentially through the URLs of all registered applications and the default, so that it must remember which is the next URL between the function callings. This is the reason why the variables are static.